File: com.issue.tracker.infra.web/src/main/java/com/issue/tracker/infra/web/common/ServletUtils.java
package com.issue.tracker.infra.web.common;

import java.io.BufferedReader;
import java.io.IOException;

public class ServletUtils {
    public static String readBody(BufferedReader reader) throws IOException {
        StringBuilder sb = new StringBuilder();
        String line;
        while ((line = reader.readLine()) != null) {
            sb.append(line);
        }
        return sb.toString();
    }
}

File: com.issue.tracker.infra.web/src/main/kotlin/com/issue/tracker/infra/web/auth/ServletUtils.kt
package com.issue.tracker.infra.web.auth

import java.io.BufferedReader

object ServletUtils {
    fun readBody(reader: BufferedReader): String {
        val sb = StringBuilder()
        var line: String?
        while (reader.readLine().also { line = it } != null) {
            sb.append(line)
        }
        return sb.toString()
    }
}